include/deal.II/lac/la_parallel_vector.h

```diff
@@ -1325,56 +1383,144 @@ namespace LinearAlgebra
 
 #ifndef DOXYGEN
 
+    namespace internal
+    {
+      template <typename Number, typename MemorySpace>
+      struct la_parallel_vector_functions
```

Comment:
This is basically an "Implementation" or "Policy" class. Maybe it's worth using that name.

I would probably have put it into the class itself. It's not useful to anyone else.

Reply:
I've tried to move the structure into the class but I can't make it work. According to [this](https://stackoverflow.com/questions/28193521/nested-class-specialization-in-terms-of-outer-class-template-arguments) there are no easy way to do a partial specialization for the nested struct. I have renamed the struct though.